packages/flutter/lib/src/material/floating_action_button_theme.dart

```diff
@@ -0,0 +1,131 @@
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Defines default property values for descendant [FloatingActionButton]
+/// widgets.
+///
+/// Descendant widgets obtain the current [FloatingActionButtonThemeData] object
+/// using `FloatingActionButtonThemeData.of(context)`. Instances of
+/// [FloatingActionButtonThemeData] can be customized with
+/// [FloatingActionButtonThemeData.copyWith].
+///
+/// Typically an [FloatingActionButtonThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.floatingActionButtonTheme].
+///
+/// All [FloatingActionButtonThemeData] properties are `null` by default.
+/// When null, the [FloatingActionButton] will use the values from [ThemeData]
+/// if they exist, otherwise it will provide its own defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+class FloatingActionButtonThemeData extends Diagnosticable {
+  /// Creates a theme that can be used for
+  /// [ThemeData.floatingActionButtonTheme].
+  const FloatingActionButtonThemeData({
+    this.backgroundColor,
+    this.foregroundColor,
+    this.elevation,
+    this.highlightElevation,
+    this.shape,
+  });
+
+  /// Color to be used for the unselected, enabled floating action buttons's
+  /// background.
+  final Color backgroundColor;
+
+  /// Color to be used for the unselected, enabled floating action buttons's
+  /// foreground.
+  final Color foregroundColor;
+
+  /// The z-coordinate to be used for the unselected, enabled floating action \
+  /// buttons's elevation foreground. .
+  final double elevation;
+
+  /// The z-coordinate to be used for the selected, enabled floating action
```

Comment:
The comment looks incorrect here, extra / and the word foreground.

Reply:
Done

Reply:
Why add 'foreground'?

Reply:
It is what its called in the FloatingActionButton constructor

Reply:
I wonder if that's a typo. Makes no sense to me unless the shadows are multi-layered themselves. I know they are but not to the end developer.

Reply:
I also agree that color and backgroundColor would be the expected names, but I also do not think it should have different names between the Widget and the WidgetTheme(Data)